File: gpt/nivel avanzado/go/productos.go
package main

import (
    "fmt"
    "sort"
)

type Producto struct {
    nombre, categoria string
    precio            int
}

func main() {
    productos := []Producto{
        {"A", "X", 30},
        {"B", "Y", 20},
        {"C", "X", 50},
    }

    var filtrados []Producto
    for _, p := range productos {
        if p.categoria == "X" {
            filtrados = append(filtrados, p)
        }
    }

    sort.Slice(filtrados, func(i, j int) bool {
        return filtrados[i].precio > filtrados[j].precio
    })

    for _, p := range filtrados {
        fmt.Printf("%s: %d\n", p.nombre, p.precio)
    }
}

File: gpt/nivel avanzado/c++/productos.cpp
#include <iostream>
#include <vector>
#include <algorithm>

struct Producto {
    std::string nombre, categoria;
    int precio;
};

int main() {
    std::vector<Producto> productos = {
        {"A", "X", 30},
        {"B", "Y", 20},
        {"C", "X", 50}
    };

    std::vector<Producto> filtrados;
    std::copy_if(productos.begin(), productos.end(), std::back_inserter(filtrados),
        [](const Producto& p) { return p.categoria == "X"; });

    std::sort(filtrados.begin(), filtrados.end(),
        [](const Producto& a, const Producto& b) { return a.precio > b.precio; });

    for (const auto& p : filtrados)
        std::cout << p.nombre << ": " << p.precio << std::endl;
}
